[Package]: neocmakelsp

### Why is it worth to add this package?

It is a CMake lsp based on tower-lsp and treesitter

### Home page URL

https://github.com/Decodetalkers/neocmakelsp

### Source code URL

https://github.com/Decodetalkers/neocmakelsp

### Packaging policy acknowledgement

- [X] I certify that I have read [Termux Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy) and understand that my request will be denied in case of violation.

### Additional information

_No response_

Failed due to compilation error:

```
Compiling neocmakelsp v0.6.20 (/data/data/com.termux/files/home/.termux-build/neocmakelsp/src)
error[E0432]: unresolved import `cmakepackage`
   --> src/utils/findpackage.rs:133:9
    |
133 | pub use cmakepackage::*;
    |         ^^^^^^^^^^^^ use of undeclared crate or module `cmakepackage`

error[E0432]: unresolved import `crate::utils::CMAKE_PACKAGES`
 --> src/search.rs:1:34
  |
1 | use crate::utils::{CMakePackage, CMAKE_PACKAGES};
  |                                  ^^^^^^^^^^^^^^ no `CMAKE_PACKAGES` in `utils`

error[E0432]: unresolved import `super::CMAKE_PACKAGES_WITHKEY`
  --> src/utils/treehelper.rs:19:5
   |
19 | use super::CMAKE_PACKAGES_WITHKEY;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `CMAKE_PACKAGES_WITHKEY` in `utils`

error[E0425]: cannot find value `CMAKE_PACKAGES` in module `utils`
 --> src/complete/findpackage.rs:6:12
  |
6 |     utils::CMAKE_PACKAGES
  |            ^^^^^^^^^^^^^^ not found in `utils`

error[E0425]: cannot find value `CMAKE_PACKAGES_WITHKEY` in module `utils`
   --> src/complete.rs:555:30
    |
555 |     let packageinfo = utils::CMAKE_PACKAGES_WITHKEY.get(package_name)?;
    |                              ^^^^^^^^^^^^^^^^^^^^^^ not found in `utils`

error[E0425]: cannot find value `CMAKE_PACKAGES_WITHKEY` in module `utils`
  --> src/jump/findpackage.rs:11:12
   |
11 |     utils::CMAKE_PACKAGES_WITHKEY.get(&input).map(|context| {
   |            ^^^^^^^^^^^^^^^^^^^^^^ not found in `utils`

warning: unused import: `Table`
 --> src/search.rs:2:59
  |
2 | use cli_table::{format::Justify, Cell, CellStruct, Style, Table};
  |                                                           ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.
warning: `neocmakelsp` (bin "neocmakelsp") generated 1 warning
error: could not compile `neocmakelsp` (bin "neocmakelsp") due to 6 previous errors; 1 warning emitted
error: failed to compile `neocmakelsp v0.6.20 (/data/data/com.termux/files/home/.termux-build/neocmakelsp/src)`, intermediate artifacts can be found at `/data/data/com.termux/files/home/.termux-build/neocmakelsp/src/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

```sh
TERMUX_PKG_HOMEPAGE=https://github.com/Decodetalkers/neocmakelsp
TERMUX_PKG_DESCRIPTION="the lsp for cmake"
TERMUX_PKG_LICENSE=MIT
TERMUX_PKG_MAINTAINER=@termux
TERMUX_PKG_VERSION=0.6.20
TERMUX_PKG_SRCURL=$TERMUX_PKG_HOMEPAGE/archive/v$TERMUX_PKG_VERSION.tar.gz
TERMUX_PKG_SHA256=b8d61d014c076db013d3a67a9633f5a4c0bf582a4c610232ce1bf36919b2c629
TERMUX_PKG_BUILD_IN_SRC=true
TERMUX_PKG_AUTO_UPDATE=true
```

--- a/packages/neocmakelsp/build.sh
+++ b/packages/neocmakelsp/build.sh
@@ -1,0 +1,9 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/Decodetalkers/neocmakelsp
+TERMUX_PKG_DESCRIPTION="a cmake lsp based on tower-lsp and treesitter"
+TERMUX_PKG_LICENSE=MIT
+TERMUX_PKG_MAINTAINER=@termux
+TERMUX_PKG_VERSION=0.7.3
+TERMUX_PKG_SRCURL=https://github.com/Decodetalkers/neocmakelsp/archive/refs/tags/v$TERMUX_PKG_VERSION.tar.gz
+TERMUX_PKG_SHA256=38468e7d9763aeec452f778d683e36c92730954310891d02a3cc26d4eb8e8455
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_AUTO_UPDATE=true

